### ISSUE TITLE
sql: insert/update statements are slow on MySQL 8.0.x

### ISSUE THREAD

**Issue by leonfancy** [ory/hydra#1397]
**Describe the bug**
It usually takes a very long time (5 ~ 10 seconds) to get an OAuth token from Hydra. After analyze with Jaeger tracing, I found it's bacuase the INSERT and UPDATE statement of MySQL is very slow.

**To Reproduce**

```
docker network create hydraguide

# I'm using MySQL 8
docker run --name hydra-mysql -p 3306:3306 \
--network hydraguide \
-e MYSQL_ROOT_PASSWORD=password \
-e MYSQL_DATABASE=hydra \
-e MYSQL_USER=leon \
-e MYSQL_PASSWORD=password \
-d mysql:8 

# Using JWT token

docker run -d --name hydra --rm -p 4444:4444 -p 4445:4445 \
    --network hydraguide \
    -e SECRETS_SYSTEM=abedfghijklmnopoqrsfdkfjdsa1234fdsafd \
    -e LOG_LEVEL=debug \
    -e OAUTH2_EXPOSE_INTERNAL_ERRORS=true \
    -e STRATEGIES_ACCESS_TOKEN=jwt \
    -e TTL_ACCESS_TOKEN=6h \
    -e URLS_CONSENT=http://localhost:8080/consent \
    -e URLS_LOGIN=http://localhost:8080/login \
    -e URLS_SELF_ISSUER=http://localhost:4444 \
    -e DSN=$DSN \
    -e TRACING_PROVIDER=jaeger \
    -e TRACING_PROVIDER_JAEGER_SAMPLING_SERVER_URL=http://jaeger:5778/sampling \
    -e TRACING_PROVIDER_JAEGER_LOCAL_AGENT_ADDRESS=jaeger:6831 \
    -e TRACING_PROVIDER_JAEGER_SAMPLING_TYPE=const \
    -e TRACING_PROVIDER_JAEGER_SAMPLING_VALUE=1 \
   oryd/hydra:v1.0.0-rc.9 serve all --dangerous-force-http
```

**Expected behavior**
Expected to have better performance.

**Screenshots**
![image](https://user-images.githubusercontent.com/4530475/56879062-a56da180-6a89-11e9-9cc6-28567e130d66.png)
![image](https://user-images.githubusercontent.com/4530475/56879089-c209d980-6a89-11e9-8825-57dae9df4905.png)

**Version:**
 - Environment: Docker
 - MySQL: 8
 - Hydra:  v1.0.0-rc.9

**Comment by aeneasr** [comment]
Thank you for the report. A slow database insert can be caused by several issues:

+ The db is too large (TBs)
+ Not enough cpu or memory (likely with local Docker)
+ Not enough disk space
+ Too much throughput (too many concurrent writes)

Which one would apply to your case? We run e2e tests against MySQL and have not experience long queries (tests would fail) so it has to be your DB deployment.

One last issue could by using MySQL 8.0.x - although I am clueless as to why that would be.

**Comment by aeneasr** [comment]
There seem to be several posts on slow inserts with MySQL 8. For example: https://dba.stackexchange.com/questions/216352/inserts-in-mysql-8-are-slower-than-inserts-in-mysql-5-7

**Comment by aeneasr** [comment]
I was unable to reproduce your findings on MySQL 8.0.x which only leaves the four options that I pointed out in this comment: https://github.com/ory/hydra/issues/1397#issuecomment-487490453

![image](https://user-images.githubusercontent.com/3372410/56888091-8401bb00-6a73-11e9-8d6d-fc5404667cd4.png)

**Comment by leonfancy** [comment]
Maybe it's because I run MySQL in docker, I'm using Docker on Windows 10. After changed to setup MySQL direclty on the host machine, this issue doens't happend.

**Comment by aeneasr** [comment]
That's probably the case then - closing :)

**Comment by ngrigoriev** [comment]
Just a related thought. I was thinking that the software like Hydra typically would issue a lot of identical DB queries - for the introspection queries, /userinfo etc. I have spotted sometihng about MySQL 8.0 - the query cache that was available in 5.7 and below has been removed (https://dev.mysql.com/doc/refman/5.7/en/query-cache.html). I wonder if that will be a negative thing for Hydra.
